### ISSUE TITLE
passing parameters to lanes in ios works but nothing is passed in android

### ISSUE THREAD

**Issue by maxkorp** [fastlane/fastlane#16518]
### New Issue Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Issue Description
Parameters to lanes work for platform `ios` but not for platform `android. In ios options passed show up in the `options` param as described here: https://docs.fastlane.tools/advanced/lanes/#passing-parameters

The same code (even just a `puts options`) under android fails, as `options` is empty (or nil or something, my ruby-fu is weak). 

It is not order specific, as reversing it (such that android is before ios) still fails for android and succeeds with ios. 

##### Command executed
`fastlane ios build app:professional environment:staging` (works)

`fastlane android build app:professional environment:staging` (fails)

##### Complete output when running fastlane, including the stack trace and command used
```
[✔] 🚀
[17:39:03]: Get started using a Gemfile for fastlane https://docs.fastlane.tools/getting-started/ios/setup/#use-a-gemfile
[17:39:04]: ----------------------------------------
[17:39:04]: --- Step: Verifying fastlane version ---
[17:39:04]: ----------------------------------------
[17:39:04]: Your fastlane version 2.148.1 matches the minimum requirement of 2.148.1  ✅
[17:39:04]: Driving the lane 'android build' 🚀
[17:39:04]: fish
[17:39:04]:
[17:39:04]: cat
+---------------+---------------+
|         Lane Context          |
+---------------+---------------+
| PLATFORM_NAME | android       |
| LANE_NAME     | android build |
+---------------+---------------+
[17:39:04]: Error in your Fastfile at line 10
[17:39:04]:      8:
[17:39:04]:      9:	def getAppSettings(options)
[17:39:04]:  => 10:	  environment = options[:environment]
[17:39:04]:     11:	  suffix = case environment
[17:39:04]:     12:	  when "production"

+------+----------------------------+-------------+
|                fastlane summary                 |
+------+----------------------------+-------------+
| Step | Action                     | Time (in s) |
+------+----------------------------+-------------+
| 1    | Verifying fastlane version | 0           |
+------+----------------------------+-------------+

[17:39:04]: fastlane finished with errors

Looking for related GitHub issues on fastlane/fastlane...

➡️  Undefined method `strip' for nil:NilClass error after a sucess upload_to_testflight
    https://github.com/fastlane/fastlane/issues/16201 [open] 4 💬
    3 weeks ago

➡️  [Swift] undefined method `each' for nil:NilClass - Cannot Create Group Within FastlaneSwiftRunner Project
    https://github.com/fastlane/fastlane/issues/15184 [open] 27 💬
    a day ago

➡️  [scan] undefined method `scan' for nil:NilClass
    https://github.com/fastlane/fastlane/issues/16480 [closed] 1 💬
    5 days ago

and 433 more at: https://github.com/fastlane/fastlane/search?q=undefined%20method%20%60%5B%5D%27%20for%20nil&type=Issues&utf8=✓

🔗  You can ⌘ + double-click on links to open them directly in your browser.
Traceback (most recent call last):
	19: from /usr/local/Cellar/fastlane/2.148.1/libexec/bin/fastlane:23:in `<main>'
	18: from /usr/local/Cellar/fastlane/2.148.1/libexec/bin/fastlane:23:in `load'
	17: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/bin/fastlane:23:in `<top (required)>'
	16: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/cli_tools_distributor.rb:119:in `take_off'
	15: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/commands_generator.rb:41:in `start'
	14: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/commands_generator.rb:352:in `run'
	13: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/commander-fastlane-4.4.6/lib/commander/delegates.rb:15:in `run!'
	12: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane_core/lib/fastlane_core/ui/fastlane_runner.rb:76:in `run!'
	11: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/commander-fastlane-4.4.6/lib/commander/runner.rb:476:in `run_active_command'
	10: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:153:in `run'
	 9: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/commander-fastlane-4.4.6/lib/commander/command.rb:178:in `call'
	 8: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/commands_generator.rb:108:in `block (2 levels) in run'
	 7: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/command_line_handler.rb:36:in `handle'
	 6: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/lane_manager.rb:47:in `cruise_lane'
	 5: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/runner.rb:45:in `execute'
	 4: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/runner.rb:45:in `chdir'
	 3: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/runner.rb:49:in `block in execute'
	 2: from /usr/local/Cellar/fastlane/2.148.1/libexec/gems/fastlane-2.148.1/fastlane/lib/fastlane/lane.rb:33:in `call'
	 1: from Fastfile:103:in `block (2 levels) in parsing_binding'
Fastfile:10:in `getAppSettings': [!] undefined method `[]' for nil:NilClass (NoMethodError)
```

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

### Environment

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                                   |
| --------------------------- | ------------------------------------------------------- |
| OS                          | 10.15.4                                                 |
| Ruby                        | 2.5.8                                                   |
| Bundler?                    | false                                                   |
| Git                         | git version 2.26.2                                      |
| Installation Source         | /usr/local/Cellar/fastlane/2.148.1/libexec/bin/fastlane |
| Host                        | Mac OS X 10.15.4 (19E287)                               |
| Ruby Lib Dir                | /usr/local/Cellar/ruby@2.5/2.5.8/lib                    |
| OpenSSL Version             | OpenSSL 1.1.1f  31 Mar 2020                             |
| Is contained                | false                                                   |
| Is homebrew                 | false                                                   |
| Is installed via Fabric.app | false                                                   |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/             |
| Xcode Version               | 11.5                                                    |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   |             |   |
| LANGUAGE |             |   |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
fastlane_version '2.148.1'

# before_all do
#   ensure_git_branch
#   ensure_git_status_clean
#   git_pull
# end

def getAppSettings(options)
  environment = options[:environment]
  suffix = case environment
  when "production"
    ""
  when "staging"
    ".test"
  else
    raise "Invalid environment #{environment}, please choose production or staging"
  end

  app = options[:app]
  case app
  when 'professional'
    {
      :dir => "./packages/mobile-professional",
      :iosName => "App1Name",
      :appId => "com.id1.id1#{suffix}",
      :ipa => "App1Name"
    }
  when 'business'
    {
      :dir => "./packages/mobile-business",
      :iosName => "App2Name",
      :appId => "com.id2.id2#{suffix}",
      :ipa => "App2Name"
    }
  else
    raise "Invalid application #{app}, please choose professional or business"
  end
end

platform :ios do
  # iOS Lanes
  # private_lane :build do
  private_lane :prebuild do |options|
    settings = getAppSettings(options)
    increment_build_number(xcodeproj: "#{settings[:dir]}/ios/#{settings[:iosName]}.xcodeproj")
    cocoapods(podfile: "#{settings[:dir]}/ios/Podfile")
    update_app_identifier(
      xcodeproj: "#{settings[:dir]}/ios/#{settings[:iosName]}.xcodeproj",
      plist_path: "#{settings[:iosName]}/Info.plist",
      app_identifier: settings[:appId]
    )
  end


  desc 'Build the iOS application.'
  lane :build do |options|
    puts('fish')
    puts(options)
    puts('cat')
    prebuild(options)
    settings = getAppSettings(options)
    gym(
      export_method: "app-store",
      workspace: "#{settings[:dir]}/ios/#{settings[:iosName]}.xcworkspace",
      scheme: settings[:iosName],
      configuration: "Release",
      include_symbols: true,
      include_bitcode: true,
      export_xcargs: "-allowProvisioningUpdates",
      output_directory: "#{settings[:dir]}/ios/export"
    )
  end

  lane :release_staging do |options|
    build
    upload_to_testflight(
      ipa: "#{settings[:dir]}/ios/export/#{settings[:ipa]}.ipa",
      skip_waiting_for_build_processing: true,
      skip_submission: true
    )
  end

end

platform :android do |options|
  # Android Lanes
  desc 'Build the Android application.'
  lane :build do
    puts('fish')
    puts(options)
    puts('cat')
    settings = getAppSettings(options)
    gradle(task: 'clean', project_dir: "#{settings[:dir]}/android/")
    gradle(task: 'assemble', build_type: 'Release', project_dir: "#{settings[:dir]}/android/")
  end

end

```
</details>

<details><summary>`./fastlane/Appfile`</summary>

```ruby
for_platform :ios do
  app_identifier "com.id1.id1"
  apple_id "{redacted}"

  for_lane :build_staging do
    app_identifier "com.id1.id1.test"
  end
end

```
</details>

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.148.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 3.0.2        |
| claide                    | 1.0.3        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.16.0       |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.7.0        |
| terminal-table            | 1.8.0        |
| plist                     | 3.5.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.7.0        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.7.1        |
| tty-cursor                | 0.7.1        |
| tty-spinner               | 0.9.3        |
| babosa                    | 1.0.3        |
| colored                   | 1.2          |
| highline                  | 1.7.10       |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.73.0       |
| faraday                   | 1.0.1        |
| unf_ext                   | 0.0.7.7      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20190701 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| faraday_middleware        | 1.0.0        |
| fastimage                 | 2.1.7        |
| gh_inspector              | 1.1.3        |
| json                      | 2.3.0        |
| mini_magick               | 4.10.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.3.0        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 1.0.0        |
| dotenv                    | 2.7.5        |
| bundler                   | 2.1.4        |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.8        |
| jwt                       | 2.1.0        |
| uber                      | 0.1.0        |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 3.1.2        |
| mini_mime                 | 1.0.2        |
| multi_json                | 1.14.1       |
| signet                    | 0.14.0       |
| memoist                   | 0.16.2       |
| os                        | 1.1.0        |
| googleauth                | 0.12.0       |
| httpclient                | 2.8.3        |
| google-api-client         | 0.38.0       |
| google-cloud-env          | 1.3.1        |
| google-cloud-errors       | 1.0.0        |
| google-cloud-core         | 1.5.0        |
| digest-crc                | 0.5.1        |
| google-cloud-storage      | 1.26.1       |
| emoji_regex               | 1.0.1        |
| jmespath                  | 1.4.0        |
| aws-partitions            | 1.317.0      |
| aws-eventstream           | 1.1.0        |
| aws-sigv4                 | 1.1.3        |
| aws-sdk-core              | 3.96.1       |
| aws-sdk-kms               | 1.31.0       |
| aws-sdk-s3                | 1.65.0       |
</details>

*generated on:* **2020-05-26**
</details>

<details>
  <pre> [REPLACE THIS WITH YOUR INFORMATION] </pre>
</details>

**Comment by maxkorp** [comment]
after some further reasearch, `ios` works, `android` does not, lanes defined outside of a platform do work, and lanes defined in an unsupported platform (e.g. `iospants` or `zandroid`) do not.

**Comment by maxkorp** [comment]
OR, or, just hear me out, I am an idiot and I put the `|options|` on the platform not on the lane. 
This may or may not have been what happened, but it definitely was.

Since I trimmed it down and copy pasted around to test other platforms rather than starting fresh (and it was syntactically valid so I didn't see an error), it continued to die. Once i tried to create a simple test by just copying the working ios platform with one test lane, I realized it. 
Sorry!
